### ISSUE TITLE
create Map<K, V>::query_all

### ISSUE THREAD

**Issue by ewoolsey** [CosmWasm/cw-storage-plus#27]
function signature should probably look something like:
```rust
impl<'a, K, T> Map<'a, K, T>
where
    T: Serialize + DeserializeOwned,
    K: PrimaryKey<'a>,
{
        pub fn query_all<Q: CustomQuery>(
        &self,
        querier: &QuerierWrapper<Q>,
        remote_contract: Addr,
        order: Order // Not sure this is required
    ) -> StdResult<Vec<T>> {
        ...
    }
}
```

The purpose of this function is to allow the querying of external contracts, and should function similarly to the existing Map::query.

**Comment by ewoolsey** [comment]
Yeah I don't think using range is possible, though I could be wrong. Either way being able to query the entire vec would be super useful as well!

**Comment by chipshort** [comment]
The problem is, there is no "entire vec" there. It's all saved individually in storage under the map's key concatenated with the key you put the values into. So, to my knowledge, the only way to get all the values out is using range or if you know all the keys and query them individually.
